### DISCUSSION TITLE
Can I have multiple services running for the same app?

### DISCUSSION THREAD

**Discussion by lemedro** [dokku/dokku#5158] [Q&A] [answered]
Hi, is there a way that I can have a few containers running for an app? Like the main server, the auth service and the recommendation service at the same time for the same app? I don't need k8s complexity and Dokku seems perfect for a lot of my use cases and I was wondering about this one. Should I create plugins, or should I create separate apps? And how do I make the services talk to each other internally, something like docker network, if I was using docker-compose?

**Comment by josegonzalez** [comment, marked as answer]
You can deploy each of the processes separately - as different apps - or together - using a `Procfile` to specify each app. Scaling is done like so: https://dokku.com/docs/processes/process-management/

For networking between processes, you can use the `attach-post-deploy` network (thats probably the one you want) as described here: https://dokku.com/docs/networking/network/#attaching-an-app-to-a-network (create a network and then attach it to each app that that phase).
